### DISCUSSION TITLE
Setting light colors?

### DISCUSSION THREAD

**Discussion by Jonathanese** [net-daemon/netdaemon#719] [General]
I'm relatively new to NetDaemon, but I'm glad I found it because it is a much better format for creating complex automations.

I'm currently trying to build a wrapper or interface or whatever the term is. Basically, I'm trying to create libraries to simplify the functionality to where you are doing more hard-typed native c# coding and less of the raw API calls.

I'm currently working on the lighting wrapper and am learning the basics so I can get an idea of how I'll organize things. I'm past "blink a light" and on to "change a color" and I'm having issues.  TurnOn() can take a object rgbColor. But I'm not sure what it's supposed to look like. I've tried a few variations in the code below. Different colors so it is clear which one succeeded.

          
```
            scheduler.Schedule(TimeSpan.FromSeconds(2), () => light.TurnOn(0));
            scheduler.Schedule(TimeSpan.FromSeconds(4), () => light.TurnOff());

            scheduler.Schedule(TimeSpan.FromSeconds(6), () => light.TurnOn(0,rgbColor: "Blue"));
            scheduler.Schedule(TimeSpan.FromSeconds(8), () => light.TurnOff());

            scheduler.Schedule(TimeSpan.FromSeconds(10), () => light.TurnOn(0,rgbColor: "128,0,0"));
            scheduler.Schedule(TimeSpan.FromSeconds(12), () => light.TurnOff());

            scheduler.Schedule(TimeSpan.FromSeconds(14), () => light.TurnOn(0,rgbColor: "[0,128,0]"));
            scheduler.Schedule(TimeSpan.FromSeconds(16), () => light.TurnOff());
```

Only TurnOn(0) turns the light on.

Ultimately here's what my function would look like.

```
public void SetColor(LightEntity light, Color color)
        {
            if(color != Color.Black)
            {
                string colorstring = $"{color.R},{color.G},{color.B}";
                light.TurnOn(rgbColor: colorstring);
            }
            else
            {
                light.TurnOff();
            }
           
        }
```

This way, instead of tossing around and converting strings, I can use a System.Drawing.Color object. Or perhaps a custom color object.

Not sure where development is on this, but I haven't seen any examples that use many of the parameters and certainly none that use the multi-value parameters, so I wasn't sure if this was implemented yet. If not, I might find another workaround to get colors to set.

**Comment by helto4real** [comment]
Please join the discord server. We have support there. Check netdaemon.xyz for invite url.
